### DISCUSSION TITLE
HTTP to HTTPS redirection?

### DISCUSSION THREAD

**Discussion by miroljub1995** [ory/hydra#3158] [Q&A] [answered]
I'm wondering if hydra supports HTTP to HTTPS redirection?
Thanks.

**Comment by vinckr** [comment, marked as answer]
Hello @miroljub1995 
I am not sure, but try it out and let us know how it goes. 

Can you also elaborate a bit more on the usecase, is that a redirection in your app - why not enforce https everywhere? 

See also [SSL/TLS, HTTPS, Self-Signed Certificates](https://www.ory.sh/docs/hydra/guides/ssl-https-tls)

**Reply by miroljub1995** [reply]
We use https, but would like to redirect to https if http://... is hit from anywhere in k8s. Now http is unreachable. One workaround is to add sidecar container (e.g. nginx) which listens 80 and redirect to 443.
I was looking for some flag in hydra to listen both http and https and do redirect from http to https, but I couldn't find such.

**Reply by vinckr** [reply]
Oh I see
> sidecar container (e.g. nginx) which listens 80 and redirect to 443.

sounds good,
What do you use to protect your Hydra admin endpoints? maybe you can just use that to do the above? we use [ory/oathkeeper](https://github.com/ory/oathkeeper) for most of that.

**Reply by miroljub1995** [reply]
We've just started with hydra and probably will include some basic NetworkPolicy at some points at a time.

**Reply by vinckr** [reply]
Hey @miroljub1995 ,
Just checking in how you proceed since we last wrote here or if there was still something that you need help with.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
